src/lib/components/chat/ChatWindow.svelte

```diff
@@ -38,6 +39,9 @@
 
 	let loginModalOpen = false;
 	let message: string;
+	let timeout: ReturnType<typeof setTimeout>;
```

Comment:
Kind of a nit or maybe I'm missing something: does the timeout needs to be referenced and handled in the `onDestroy` and can't we just do it like this?
```
on:click={() => {
	dispatch("share");
	isSharedRecently = true;
	setTimeout(() => {
		isSharedRecently = false;
	}, 2000);
}}
```

just curious, other than that lgtm

Reply:
> does the timeout needs to be referenced and handled in the onDestroy and can't we just do it like this?

yes, for the edge case.

If a user clicks `share btn` & closes the window within less than 2 seconds, it would end up creating memory leak